### ISSUE TITLE
TemporalFeature and its usage

### ISSUE THREAD

**Issue by Airy59** [Interoperable-data/ERA-Ontology-3.1.0#9]
Introducing TemporalFeature, as a subclass to time:TemporalEntity and time:TemporalDuration, is a most interesting move.

The way it is (or is not) used raises questions.

In the case of topology: only Navigability has a property era:validity with range era:TemporalFeature. This should also apply to net elements (topology is not frozen for all times, and handling successive versions from inside the model is recommended). Maybe a common superclass for both NetElement and Navigability would make sense; it could be TopologicalObject provided that the locations are managed independently from topology (see other issue).

In the case of technical characteristics: you use era:Applicability as a subclass to establish a relationship between an infra object and its technical characteristics. This is broadly correct, and expresses the fact that a technical characteristic may be time-dependent. But, getting into more detail:

-  era:TechnicalCharacteristic in the diagram seems to be era:TechnicalCharacteristic**s** in the model (a container class, so to say).
- what varies with time is, usually, the value of the characteristics
Bottom line, won't it be more appropriate to subclass characteristics **values** from TemporalFeature?

The packaging of characteristics is well in line with yearly RINF releases, but if update frequency changes to daily, that bundling may become a burden.

**Comment by remiColletGG** [comment]
I am fairly certain that Airy's comment related to a previous version of the model than the one that is currently available. However, since my questions is on the same topic, I will add to this issue. 

Reference : time ontology https://www.w3.org/TR/owl-time/#

If I am not mistaken, the following diagram is a summary of how temporal things could be handled.

![Image](https://github.com/user-attachments/assets/108698c5-efaa-46a8-9d82-410755998e49)

`era:TemporalFeature` is a "cousin" of  `time:Interval` and `time:Instant`. I am unsure of how the subclassing of `era:TemporalFeature` is really useful in integrating the time ontology to ERA's data model ?

There exists a `time:hasTime` property that could relate any `era:Feature` to a `time:TemporalEntity`

Consider the following :

![Image](https://github.com/user-attachments/assets/ecb5e81a-b546-4ae5-8315-d9ee69869f80)

Other ontologies (see some of them here https://lov.linkeddata.es/dataset/lov/vocabs/time) have had a similar approach.

What do you think of this ?

**Comment by gatemezing** [comment]
Dear @remiColletGG , thank you for the explanations.
Based on the different bilaterals and workshops, the ontology has evolved. See the new pictures (as of 25-02-2025) can be available at the following links:
- https://linkedvocabs.org/data/era-ontology/3.1.0/appGuide/resources/topology-v3.1-20250227.png  
- https://linkedvocabs.org/data/era-ontology/3.1.0/appGuide/resources/ERAontology-v3.1-20250227.png. 

We have the following axiom for `era:TemporalFeature` defined as subClass of the union of both `time:TemporalDuration` and `time:TemporalEntity`. 

```
###  http://data.europa.eu/949/TemporalFeature
era:TemporalFeature rdf:type owl:Class ;
                    rdfs:subClassOf [ rdf:type owl:Class ;
                                      owl:unionOf ( time:TemporalDuration
                                                    time:TemporalEntity
                                                  )
                                    ] ;
                    dcterms:created "2024-05-24"^^xsd:date ;
                    rdfs:comment "The union of TemporalDuration that represents a time extent and TemporalEntity that represents a temporal interval or instant." ;
                    rdfs:isDefinedBy <http://data.europa.eu/949/> ;
                    rdfs:label "Temporal Feature"@en .
```
That gives us the flexibility (IMHO) to use more properties of both classes that just using the property `time:hasTime` as the link suggested in your last figure.  One use case is the parameter applicability says for 2 months. 
Note: we are planning the migration to Gitlab as announced in the readme. It is taking more time than expected.

**Comment by remiColletGG** [comment]
Thanks for the explanation ! It is true that, while most likely suboptimal, we will have to deal with temporal duration as parameter applicability. 

As you said, the name of the game is flexibility, hence I totally agree with your design choice.

**Comment by gatemezing** [comment]
Thank you @remiColletGG ! Maybe we could ask if this issue should be closed. What do you think dear @Airy59 ?
